File: apps/api/internal/files/domain/file.go
package domain

import "time"

type UploadStatus string

const (
	UploadStatusPending  UploadStatus = "pending"
	UploadStatusUploaded UploadStatus = "uploaded"
	UploadStatusFailed   UploadStatus = "failed"
)

func (s UploadStatus) IsValid() bool {
	switch s {
	case UploadStatusPending, UploadStatusUploaded, UploadStatusFailed:
		return true
	default:
		return false
	}
}

type File struct {
	ID           string
	UserID       string
	Bucket       string
	ObjectPath   string
	OriginalName string
	MimeType     string
	SizeBytes    int64
	UploadStatus UploadStatus
	CreatedAt    time.Time
}

File: apps/web/src/features/files/types/file.ts

export interface FileResource {
  id: string
  original_name: string
  mime_type: string
  size_bytes: number
  upload_status: string
  created_at: string
  uploaded_at: string | null
}

export interface UploadFileResponse extends FileResource {
  already_existed: boolean
}

export interface FileUrlResponse {
  url: string
}